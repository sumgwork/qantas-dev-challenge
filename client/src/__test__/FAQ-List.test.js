import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJSON from "enzyme-to-json";

import FAQList from "../components/FAQ-List";

configure({ adapter: new Adapter() });

const mockFAQs = () => [
  { title: "title1", body: "body1" },
  { title: "title2", body: "body2" }
];

describe("<FAQList />", () => {
  let mockSelectFAQFn = jest.fn();
  const wrapper = shallow(
    <FAQList selectedIndex={0} faqs={mockFAQs()} selectFAQ={mockSelectFAQFn} />
  );

  it("renders properly and matches snapshot", () => {
    expect(toJSON(wrapper.find("div"))).toMatchSnapshot();
  });
  it("should show correct number of faq title elements", () => {
    expect(wrapper.find("h2")).toHaveLength(mockFAQs().length);
  });
  it("should render the titles properly", () => {
    mockFAQs().forEach((faq, index) => {
      wrapper.find(".clickable").at(index);
      expect(
        wrapper
          .find(".clickable")
          .at(index)
          .text()
      ).toBe(mockFAQs()[index].title);
    });
  });

  it("should have only 1 FAQ selected", () => {
    expect(wrapper.find(".selected")).toHaveLength(1);
  });
  it("should call a function when a FAQ is clicked", () => {
    wrapper
      .find(".clickable")
      .at(1)
      .simulate("click");
    expect(mockSelectFAQFn).toHaveBeenCalledTimes(1);
  });
});
