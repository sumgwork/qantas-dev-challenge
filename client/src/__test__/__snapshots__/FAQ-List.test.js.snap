// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQList /> renders properly and matches snapshot 1`] = `
<div>
  <styled.div
    className="selected clickable"
    key="0"
    onClick={[Function]}
  >
    <h2>
      title1
    </h2>
  </styled.div>
  <styled.div
    className=" clickable"
    key="1"
    onClick={[Function]}
  >
    <h2>
      title2
    </h2>
  </styled.div>
</div>
`;
